### ISSUE TITLE
[Feature Request]: AppX or similar (appxbundle,etc) file for install on Xbox dev mode consoles.

### ISSUE THREAD

**Issue by Lux-pogu** [FreeTubeApp/FreeTube#4769]
### Guidelines

- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the feature request I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.
- [X] This issue contains only one feature request.

### Problem Description

No AppX or Appxbundle file available 

### Proposed Solution

Add a AppX or Appxbundle/similar 

### Alternatives Considered

Anything else that will work on Xbox (or dev mode)

### Issue Labels

new feature

**Comment by absidue** [comment]
Can Xboxes run apps compiled for Windows? Additionally can it be done from free or do Xbox's require you to pay Microsoft a lot of money every year to sign it? If not I don't think this will be possible, as I've just done a bit of research and can't find anything about running Electron apps on an Xbox.

I don't have an Xbox or any experience with Xbox apps or the development of them. If you are lucky maybe someone in the community knows something about it and could help you.

Also FreeTube has no controller support, as it is a desktop app not a console app, so you would be required to use a keyboard and mouse to use the app.

**Comment by Lux-pogu** [comment]
> Can Xboxes run apps compiled for Windows? Additionally can it be done from free or do Xbox's require you to pay Microsoft a lot of money every year to sign the app? If not I don't think this will be possible, as I've just done some research and can't find anything about running Electron apps on an Xbox.
> 
> I don't have an Xbox or any experience with Xbox apps or the development of them. If you are lucky maybe someone in the community knows something about it and could help you.
> 
> Also FreeTube has no controller support, as it is a desktop app not a console app, so you would be required to use a keyboard and mouse to use the app.

yes,xboxes can run AppX and Appxbundle files with the paid but not subscription (one time payment). and you don't have to sign it. since dev mode allows you to side load any AppX file. you're more so looking for if electron apps can be run in AppX files. as for no controller support,yeah I didn't know that. *maybe* converting it to m/k input it would work. but yeah based on that,without a full rewrite,it sounds unlikely.

**Comment by Onotseike** [comment]
Hello @absidue & @Lux-pogu , could this tool be useful in converting an electron app to an appx one [electron-windows-store](https://github.com/electron-userland/electron-windows-store)?

**Comment by Lux-pogu** [comment]
Yes, looks like it's made for it!

**Comment by Lux-pogu** [comment]
> Hello @absidue & @Lux-pogu , could this tool be useful in converting an electron app to an appx one [electron-windows-store](https://github.com/electron-userland/electron-windows-store)? 
> 

Looks that way!

**Comment by Onotseike** [comment]
Hello could we discuss the possibility of seeing if this works and testing with the mind to release this for Xbox?

**Comment by absidue** [comment]
Even if you (someone with an xbox would have to implement and test it) manage to package it for xbox, installing it would require activating the paid developer mode on your xbox, as it will be unsigned and not in the Microsoft Store, just like all the other FreeTube packages. Additionally as mentioned above FreeTube has no controller support, keyboard-only usage is already very limited so you would have to have a keyboard and mouse connected to your xbox to use FreeTube.

So to summarise unless someone in the community that has an xbox decides to implement xbox and controller support and also maintain it to keep it working with future changes, it is very unlikely to happen.

**Comment by Onotseike** [comment]
Okay. I will review if I have capacity to test out xbox controller support

**Comment by absidue** [comment]
You need to implement controller support in FreeTube first, before you can test anything, you'll want to take a look at the Web Gamepad API: https://developer.mozilla.org/en-US/docs/Web/API/Gamepad_API

**Comment by Onotseike** [comment]
Are there restrictions on the npm packages i can use? For example can i use [robotjs](https://www.npmjs.com/package/robotjs)
@absidue ?

**Comment by TBNRBERRY** [comment]
I would be interested in working on this project, I cannot guarantee anything but I am willing to work on this. I have made and worked on a few of my own apps and installed them on my own Xbox Series X with controller support. @Onotseike @absidue @Lux-pogu

**Comment by Onotseike** [comment]
@TBNRBERRY  could we schedule a call about the possiblity of working on this over the dec period?

**Comment by TBNRBERRY** [comment]
Sounds good, we can plan a discord call or a call later today, I will be able to talk on a call after 1:30PM CST @Onotseike

**Comment by Onotseike** [comment]
> Sounds good, we can plan a discord call or a call later today, I will be able to talk on a call after 1:30PM CST [@Onotseike](https://github.com/Onotseike)

Great, can you reach out to me via discord. I use same same name as above

**Comment by absidue** [comment]
If the two of you work on this, please keep in mind that controller support will have to be done properly not some hack simulating mouse and keyboard events with robotjs.

We the FreeTube team won't be building or publishing an xbox compatible package of FreeTube, a platform that we have no way of maintaining or testing for, that's not to say you can't create custom unofficial builds yourself, just that we won't be building those packages or providing support for any issues that occur when running FreeTube on unsupported platforms. If you decide to release your custom unofficial builds somewhere, makes sure it is clear to your users that your project has nothing to do with us and they need to come to you with all of their problems (we are a tiny team of volunteers and do not have the capacity to provide support for unofficial builds).

**Comment by TBNRBERRY** [comment]
@Onotseike @absidue I can schedule a call on discord later
